keps/976-plain-pods/README.md

```diff
@@ -0,0 +1,672 @@
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+#### Story 1
+
+As an platform developer, I can queue plain Pods. I just add a queue name to the Pods through a
+label.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: foo
+  namespace: pod-namespace
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+spec:
+  containers:
+  - name: job
+    image: hello-world
+    resources:
+      requests:
+        cpu: 1m
+```
+
+#### Story 2
+
+As a platform developer, I can queue groups of Pods that might or might not have the same shape
+(Pod specs).
+In addition to the queue name, I can specify how many Pods belong to the group.
+
+The pods of a job following a driver-workers paradigm would look like follows:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: job-driver
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/pod-group-name: pod-group
+  annotations:
+    kueue.x-k8s.io/pod-group-total-count: "3"
+spec:
+  containers:
+  - name: job
+    image: driver
+    resources:
+      requests:
+        cpu: 1m
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: job-worker-0
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/pod-group-name: pod-group
+  annotations:
+    kueue.x-k8s.io/pod-group-total-count: "3"
+spec:
+  containers:
+  - name: job
+    image: worker
+    args: ["--index", "0"]
+    resources:
+      requests:
+        cpu: 1m
+        vendor.com/gpu: 1
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: job-worker-1
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/pod-group-name: pod-group
+  annotations:
+    kueue.x-k8s.io/pod-group-total-count: "3"
+spec:
+  containers:
+  - name: job
+    image: worker
+    args: ["--index", "1"]
+    resources:
+      requests:
+        cpu: 1m
+        vendor.com/gpu: 1
+```
+
+### Story 3
+
+Motivation: In frameworks like Spark, worker Pods are only created after by the driver Pod. As
+such, the worker Pods specs cannot be predicted beforehand. Even though the job could be considered
+elastic, generally users wouldn't want to start a spark driver to run if no workers would fit.
+
+As a Spark user, I can queue the driver Pod while providing the expected shape of the worker Pods.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: job-driver
+  labels:
+    kueue.x-k8s.io/queue-name: user-queue
+    kueue.x-k8s.io/pod-group-name: pod-group
+  annotations:
+    # If the template is left empty, it means that it will match the spec of this pod.
+    kueue.x-k8s.io/pod-group-sets: |-
+      [
+        {
+          name: driver,
+          count: 1,
```

Comment:
If the driver count is 3, then we'll create two more pods?  Generally it seems a little weird as if we're implementing a controller for this kind of job and increase the complexity a lot. Maybe start with the first two stories?